src/main/kotlin/com/exactpro/th2/act/ActHandlerKt.kt

```diff
@@ -0,0 +1,389 @@
+            val expectedValue = requireNotNull(request.message.fieldsMap["ClOrdID"])
+            val settings = CallSettings("placeOrderMassCancelRequestFIX", expectedValue.simpleValue, mapOf("OrderMassCancelReport" to "ClOrdID"))
+            placeMessage(settings, request, responseObserver)
+        }.onFailure {
+            LOGGER.error(it) { "Failed to place an OrderMassCancelRequest. Message = ${request.message}" }
+            responseObserver.sendErrorResponse("Failed to place an OrderMassCancelRequest. Error: ${it.message}")
+        }.onSuccess {
+            LOGGER.debug { "placeOrderMassCancelRequestFIX successfully finished" }
+        }
+    }
+
+    override fun placeQuoteCancelFIX(
+        request: PlaceMessageRequest,
+        responseObserver: StreamObserver<PlaceMessageResponse>
+    ) {
+        runCatching {
+            LOGGER.debug { "placeQuoteCancelFIX request: $request" }
+            val expectedValue = requireNotNull(request.message.fieldsMap["QuoteMsgID"])
+            val settings = CallSettings("placeQuoteCancelFIX", expectedValue.simpleValue, mapOf("MassQuoteAcknowledgement" to "QuoteID"))
+            placeMessage(settings, request, responseObserver)
+        }.onFailure {
+            LOGGER.error(it) { "Failed to place an QuoteCancel. Message = ${request.message}" }
+            responseObserver.sendErrorResponse("Failed to place a QuoteCancel. Error: ${it.message}")
+        }.onSuccess {
+            LOGGER.debug { "placeQuoteCancelFIX successfully finished" }
+        }
+    }
+
+    override fun placeQuoteRequestFIX(
+        request: PlaceMessageRequest,
+        responseObserver: StreamObserver<PlaceMessageResponse>
+    ) {
+        runCatching {
+            LOGGER.debug { "placeQuoteRequestFIX request: $request" }
+            val expectedValue = requireNotNull(request.message.fieldsMap["QuoteReqID"])
+            val settings = CallSettings("placeQuoteRequestFIX", expectedValue.simpleValue, mapOf("QuoteStatusReport" to "QuoteReqID"))
+            placeMessage(settings, request, responseObserver)
+        }.onFailure {
+            LOGGER.error(it) { "Failed to place an QuoteRequest. Message = ${request.message}" }
+            responseObserver.sendErrorResponse("Failed to place a QuoteRequest. Error: ${it.message}")
+        }.onSuccess {
+            LOGGER.debug { "placeQuoteRequestFIX successfully finished" }
+        }
+    }
+
+    override fun placeQuoteResponseFIX(
+        request: PlaceMessageRequest,
+        responseObserver: StreamObserver<PlaceMessageResponse>
+    ) {
+        runCatching {
+            LOGGER.debug { "placeQuoteResponseFIX request: $request" }
+            val expectedValue = requireNotNull(request.message.fieldsMap["RFQID"])
+            val settings = CallSettings("placeQuoteResponseFIX", expectedValue.simpleValue, mapOf("ExecutionReport" to "RFQID", "QuoteStatusReport" to "RFQID"))
+            placeMessage(settings, request, responseObserver)
```

Comment:
This part is different for calls, but other code in this method looks like a boilerplate.
Maybe create a common method and pass a lambda to extract the expected field into it.